idea/idea-frontend-fir/src/org/jetbrains/kotlin/idea/frontend/api/fir/components/KtFirExpressionTypeProvider.kt

```diff
@@ -50,6 +52,18 @@ internal class KtFirExpressionTypeProvider(
         }
     }
 
+    override fun getPsiTypeForKtExpression(
```

Comment:
Let's extract `getPsiTypeForKtExpression` and `getPsiType(ktTypeReference: KtTypeReference..)` into separate component, smth like `KtPsiTypeProvider` and move all firType -> psiType mapping from light classes to it. And just use those new functions from light classes. This way we get rid of cyclic dependency between light classes and HL API

Reply:
Extracted those utils to `KtPsiTypeProvider` as per the suggestion. See the last commit.

Re: cyclic dependecy,

actually, `asPsiType` is inside the same module, i.e., not in light classes, but in idea-frontend-fir. Simply it was my mistake to open that as `public`; `internal` should be good enough, and changed that way.